jquery.overlay.js

```diff
@@ -211,23 +216,40 @@
           // Style attribute's string
           style = 'background-color:' + strategy.css['background-color'];
 
-          text.contents().each(function () {
-            var text, html, str, prevIndex;
-            if (this.nodeType != Node.TEXT_NODE) return;
-            text = this.textContent;
+          if (this.allowOverlapping) {
+            var textContent, prevIndex, str, html;
+            textContent = text.html()
             html = '';
             for (prevIndex = match.lastIndex = 0;; prevIndex = match.lastIndex) {
-              str = match.exec(text);
+              str = match.exec(textContent);
               if (!str) {
-                if (prevIndex) html += escape(text.substr(prevIndex));
+                if (prevIndex) html += textContent.substr(prevIndex);
                 break;
               }
               str = str[0];
-              html += escape(text.substr(prevIndex, match.lastIndex - prevIndex - str.length));
-              html += '<span style="' + style + '">' + escape(str) + '</span>';
+              html += textContent.substr(prevIndex, match.lastIndex - prevIndex - str.length);
+              html += '<span style="' + style + '">' + str + '</span>';
             };
-            if (prevIndex) $(this).replaceWith(html);
-          });
+            if (prevIndex) text.html(html);
+          } else {
+            text.contents().each(function () {
+              var text, html, str, prevIndex;
+              if (this.nodeType != Node.TEXT_NODE) return;
+              text = this.textContent;
+              html = '';
+              for (prevIndex = match.lastIndex = 0; ; prevIndex = match.lastIndex) {
```

Comment:
this empty end condition makes me fear for infinite loops.  I guess that generally shouldn't happen because as you call $(this).replaceWith, you should eventually get to a point where match.exec(text) returns false?  It seems risky to make the entire application's stability rely on that bit of logic (what if somehow escape(str) were inserting something that matched again?) - seems like we should have a fallback end condition for the loop or some sort of demonstrable guarantee that this loop will exit.